apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx

```diff
@@ -53,38 +60,211 @@ export function OrderTable({ orders }: { orders: AllSwapsQuery["swaps"] }) {
   const txUrl = buildBlockExplorerTxUrl({
-    chainId: order.chainId,
-    txHash: order.transactionHash,
+    chainId: transaction.swaps[0].chainId,
+    txHash: transaction.id,
   });
 
+  const [showOrdersRows, setShowOrdersRows] = useState(false);
+
+  const equalTokensIn = transaction.swaps.every(
+    (swap) => swap.tokenIn?.id == transaction.swaps[0].tokenIn?.id,
+  );
+
+  const equalTokensOut = transaction.swaps.every(
+    (swap) => swap.tokenOut?.id == transaction.swaps[0].tokenOut?.id,
+  );
+
+  const decimalsTokenIn = transaction.swaps[0].tokenIn?.decimals || 18;
+
+  const totalAmountTokenIn = transaction.swaps.reduce(
+    (acc, swap) =>
+      acc + Number(formatUnits(swap.tokenAmountIn, decimalsTokenIn)),
+    0,
+  );
+
+  return (
+    <>
+      <Table.BodyRow key={transaction.id}>
+        <Table.BodyCell>
+          <button onClick={() => setShowOrdersRows(!showOrdersRows)}>
+            {showOrdersRows ? (
+              <ArrowUpIcon className="h-5 w-5 text-blue9 hover:text-blue10" />
+            ) : (
+              <ArrowDownIcon className="h-5 w-5 text-blue9 hover:text-blue10" />
+            )}
+          </button>
+        </Table.BodyCell>
+        <Table.BodyCell>
+          {equalTokensIn ? (
+            <TokenInfo
+              id={transaction.swaps[0].tokenIn?.id}
+              symbol={transaction.swaps[0].tokenIn?.symbol}
+              chainId={transaction.swaps[0].chainId}
+            />
+          ) : (
+            "Multiple Tokens"
+          )}
+        </Table.BodyCell>
+        <Table.BodyCell>
+          {equalTokensIn
+            ? formatNumber(totalAmountTokenIn, 4, "decimal", "standard", 0.0001)
+            : "Multiple Tokens"}
+        </Table.BodyCell>
+        <Table.BodyCell>
+          {equalTokensOut ? (
+            <TokenInfo
+              id={transaction.swaps[0].tokenOut?.id}
+              symbol={transaction.swaps[0].tokenOut?.symbol}
+              chainId={transaction.swaps[0].chainId}
+            />
+          ) : (
+            "Multiple Tokens"
+          )}
+        </Table.BodyCell>
+        <Table.BodyCell>
+          <div className="flex items-center gap-x-1">
+            <span>{transactionStatus}</span>
+            {txUrl && (
+              <Link href={txUrl} target="_blank">
+                <ArrowTopRightIcon className="hover:text-slate11" />
+              </Link>
+            )}
+          </div>
+        </Table.BodyCell>
+        <Table.BodyCell>
+          <CancelButton
+            disabled={[
+              TransactionStatus.EXECUTED,
+              TransactionStatus.CANCELED,
+              TransactionStatus.EXECUTED_AND_CANCELED,
+            ].includes(transactionStatus)}
+          />
+        </Table.BodyCell>
+      </Table.BodyRow>
+      {showOrdersRows &&
+        transaction.swaps.map((order, index) => (
+          <TableRowOrder
+            key={order.id}
+            order={order}
+            orderStatus={swapStatus[index]}
+          />
```

Comment:
not a fan of using index 🤔

Reply:
yeah, I get your point, this can lead to some errors. I'll try to create one object that has all the data that we need.